core/CronArchive.php

```diff
@@ -1263,4 +1276,14 @@ public function checkNoDanglingInvalidations(array $idInvalidations)
             ]);
         }
     }
+
+    private function siteExists($idSite)
+    {
+        try {
+            $site = APISitesManager::getInstance()->getSiteFromId($idSite);
```

Comment:
btw could use `new Site($idSite)` here directly as it would use cache but maybe that be actually the problem and we don't want to hit the cache.

Reply:
actually be great if this hit the cache maybe? or would it cause a problem if the site was suddenly removed?

Reply:
The cache would exist for the entire core:archive run, so a site could be removed say 30 mins after archiving starts, but it's data could still be cached, and we'd try to process it. We could maybe clear the cache before starting a new site archive so it's cached a bit... I'll think about it.